src/vs/workbench/browser/web.main.ts

```diff
@@ -245,16 +244,29 @@ export class BrowserMain extends Disposable {
 		const environmentService = new BrowserWorkbenchEnvironmentService(workspace.id, logsPath, this.configuration, productService);
 		serviceCollection.set(IBrowserWorkbenchEnvironmentService, environmentService);
 
-		// Log
-		const logLevel = getLogLevel(environmentService);
-		const bufferLogger = new BufferLogger(logLevel);
-		const otherLoggers: ILogger[] = [new ConsoleLogger(logLevel)];
+		// Logger
+		const loggerService = new BrowserLoggerService(getLogLevel(environmentService), logsPath);
+		serviceCollection.set(ILoggerService, loggerService);
+
+		// Log Service
+		const otherLoggers: ILogger[] = [new ConsoleLogger(loggerService.getLogLevel())];
 		if (environmentService.isExtensionDevelopment && !!environmentService.extensionTestsLocationURI) {
-			otherLoggers.push(new ConsoleLogInAutomationLogger(logLevel));
+			otherLoggers.push(new ConsoleLogInAutomationLogger(loggerService.getLogLevel()));
 		}
-		const logService = new LogService(bufferLogger, otherLoggers);
+		const logger = loggerService.createLogger(environmentService.logFile, { id: windowLogId, name: localize('rendererLog', "Window") });
+		const logService = new LogService(logger, otherLoggers);
 		serviceCollection.set(ILogService, logService);
 
+		// Files
+		const fileService = this._register(new FileService(logService));
+		serviceCollection.set(IWorkbenchFileService, fileService);
+		// Cycling dependency between log service and file service
+		loggerService.acquireFileService(fileService);
```

Comment:
I am not a fan of expanding this pattern of debt to more places...

Reply:
Me neither, hence added you ;). 

Tried another approach to avoid cyclic dependency. 

Please check